datahub-web-react/src/app/homeV3/template/components/addModuleMenu/useAddModuleMenu.tsx

```diff
@@ -1,50 +1,134 @@
+import { AddModuleHandlerInput } from '@app/homeV3/template/types';
+
+import { PageModuleFragment } from '@graphql/template.generated';
+import { DataHubPageModuleType, EntityType, PageModuleScope } from '@types';
+
+const YOUR_ASSETS_MODULE: PageModuleFragment = {
+    urn: 'urn:li:dataHubPageModule:your_assets',
+    type: EntityType.DatahubPageModule,
+    properties: {
+        name: 'Your Assets',
+        type: DataHubPageModuleType.OwnedAssets,
+        visibility: { scope: PageModuleScope.Global },
+        params: {},
+    },
+};
+
+const DOMAINS_MODULE: PageModuleFragment = {
+    urn: 'urn:li:dataHubPageModule:top_domains',
+    type: EntityType.DatahubPageModule,
+    properties: {
+        name: 'Domains',
+        type: DataHubPageModuleType.Domains,
+        visibility: { scope: PageModuleScope.Global },
+        params: {},
+    },
+};
 
 export default function useAddModuleMenu(
     modulesAvailableToAdd: ModulesAvailableToAdd,
-    onClick?: (module: ModuleInfo) => void,
+    position: AddModuleHandlerInput,
+    closeMenu: () => void,
 ): MenuProps {
-    const convertModule = useCallback(
-        (module: ModuleInfo): ItemType => ({
-            title: module.name,
-            key: module.key,
-            label: <ModuleMenuItem module={module} />,
-            onClick: () => onClick?.(module),
-        }),
-        [onClick],
+    const { addModule } = usePageTemplateContext();
+
+    const handleAddExistingModule = useCallback(
+        (module: PageModuleFragment) => {
+            addModule({
+                module: module as PageModuleFragment,
+                position,
+            });
+            closeMenu();
+        },
+        [addModule, position, closeMenu],
     );
 
+    // TODO: use this commented out code later once we implement creating new modules
+    // const handleCreateNewModule = useCallback(
+    //     (type: DataHubPageModuleType, name: string) => {
+    //         createModule({
+    //             name,
+    //             type,
+    //             position,
+    //         });
+    //         closeMenu();
+    //     },
+    //     [createModule, position, closeMenu],
+    // );
+
     return useMemo(() => {
         const items: MenuProps['items'] = [];
 
-        if (modulesAvailableToAdd.customModules.length) {
-            items.push({
-                key: 'customModulesGroup',
-                label: <GroupItem title="Custom" />,
-                type: 'group',
-                children: modulesAvailableToAdd.customModules.map(convertModule),
-            });
-        }
+        const quickLink = {
+            title: 'Quick Link',
+            key: 'quick-link',
+            label: <MenuItem description="Choose links that are important" title="Quick Link" icon="LinkSimple" />,
+            onClick: () => {
+                // TODO: open up modal to add a quick link
+            },
+        };
 
```

Comment:
decided to hardcode these here so we can have easy onClick handlers here instead of in a constants class. open to changing this up though